### ISSUE TITLE
Allow to provide Maven gav for java.project.referencedLibraries

### ISSUE THREAD

**Issue by apupier** [redhat-developer/vscode-java#1851]
it is possible to mention path to jar on the filesystem for `java.project.referencedLibraries`, it would be nice to be able to specify Maven gav too and VS Code java will download them in the ;m2/repository and point to them automatically

```
"java.project.referencedLibraries": {
    "include": [
        "library/**/*.jar",
        "/home/username/lib/foo.jar",
        "mvn:org.example:myArtifact:1.0.0"
    ],
    "exclude": [
        "library/sources/**"
    ],
    "sources": {
        "library/bar.jar": "library/sources/bar-src.jar"
    }
}
```
sort of the corresponding feature of "Add Library -> Maven managed dependencies" in Eclipse desktop and project Settings -> Libraries in IntelliJ

##### Environment
- Operating System:
- JDK version:
- Visual Studio Code version:
- Java extension version:

##### Steps To Reproduce
1. [step 1]
2. [step 2]

[Please attach a sample project reproducing the error]
[Please attach logs](https://github.com/redhat-developer/vscode-java/wiki/Troubleshooting#enable-logging)

**Comment by fbricon** [comment]
yeah let's invent yet another build system. Then someone will ask about downloading transitive dependencies. And ask for configuring remote repos.
 
I'd rather recommend users to use JBang instead

**Comment by apupier** [comment]
the `java.project.referencedLibraries` is already about providing another build system, just a first simple piece.

use case was for Camel K, it is effectively using yet another build system and there is no plan to merge with Jbang.
